sdk/src/Core/Amazon.Runtime/Pipeline/RetryHandler/RetryPolicy.cs

```diff
@@ -446,27 +445,28 @@ private bool IsClockskew(IExecutionContext executionContext, Exception exception
                     // using accurate server time, calculate correction if local time is off
                     serverTime = serverTime.ToUniversalTime();
                     var diff = correctedNow - serverTime;
+                    var newCorrection = serverTime - realNow;
                     var absDiff = diff.Ticks < 0 ? -diff : diff;
-                    if (absDiff > clockSkewMaxThreshold)
-                    {
-                        var newCorrection = serverTime - realNow;
-                        Logger.InfoFormat(clockSkewMessageFormat,
-                            realNow, correctedNow, clientConfig.ClockOffset, serverTime, endpoint);
-
-                        // Always set the correction, for informational purposes
-                        CorrectClockSkew.SetClockCorrectionForEndpoint(endpoint, newCorrection);
 
-                        var shouldRetry = AWSConfigs.CorrectForClockSkew && !AWSConfigs.ManualClockCorrection.HasValue;
+                    Logger.InfoFormat(clockSkewMessageFormat,
+                        realNow, correctedNow, clientConfig.ClockOffset, serverTime, endpoint);
 
-                        // Only retry if clock skew correction is not disabled
-                        if (shouldRetry)
+                    // Always set the correction, for informational purposes
+                    CorrectClockSkew.SetClockCorrectionForEndpoint(endpoint, newCorrection);
+                    var shouldRetry = AWSConfigs.CorrectForClockSkew && !AWSConfigs.ManualClockCorrection.HasValue;
+                    // Some services such as S3 will return a generic 403 error for clock skew errors on HEAD requests
```

Comment:
Is this from JAVA SDK?

Reply:
If you're referring to the HEAD request comment, that was already there and is because of a COE from a while back where a HEAD request was not returning an error code since there is no body